Notation.Plugin.AzureKeyVault/Certificate/Pkcs12.cs

```diff
@@ -0,0 +1,51 @@
+using System.Security.Cryptography.Pkcs;
+using Notation.Plugin.Protocol;
+
+namespace Notation.Plugin.AzureKeyVault.Certificate
+{
+    static class Pkcs12
+    {
+        /// <summary>
+        /// Re-encode the PKCS12 data to removed the MAC and keys.
+        /// The macOS doesn't support PKCS12 with non-encrypted MAC.
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        /// <exception cref="ValidationException"></exception>
+        public static byte[] ReEncode(byte[] data)
+        {
+            Pkcs12Info pfx = Pkcs12Info.Decode(data, out _);
+            // only remove the MAC if it is password protected
+            if (pfx.IntegrityMode != Pkcs12IntegrityMode.Password)
+            {
+                return data;
+            }
+            // verify the MAC with null password
+            if (!pfx.VerifyMac(null))
+            {
+                throw new ValidationException("Invalid MAC or the MAC password is not null");
+            }
+
+            // re-build PFX without MAC
+            Pkcs12Builder pfxBuilder = new Pkcs12Builder();
+            foreach (var safeContent in pfx.AuthenticatedSafe)
+            {
+                // decrypt with null password
```

Comment:
From your code, it is empty password instead of null password. Which one is correct?

Reply:
We need null password. updated to use `ReadOnlySpan<byte>.Empty`